test/e2e/reporters/progress.go

```diff
@@ -22,22 +22,19 @@ import (
 	"fmt"
 	"io"
 	"net/http"
-	"strings"
 	"time"
 
 	"k8s.io/klog/v2"
-
-	"github.com/onsi/ginkgo/config"
-	"github.com/onsi/ginkgo/types"
 )
 
 // ProgressReporter is a ginkgo reporter which tracks the total number of tests to be run/passed/failed/skipped.
 // As new tests are completed it updates the values and prints them to stdout and optionally, sends the updates
 // to the configured URL.
+// TODO: Number of test specs is not available now, we can add it back when this is fixed in the Ginkgo V2.
+// pls see: https://github.com/kubernetes/kubernetes/issues/109744
 type ProgressReporter struct {
```

Comment:
Rather than copying behavior out of the progressreporter into various Before/After hooks, why not keep them in functions attached to the reporter and call them from the hooks? - That way we keep the string-y parts together and keep things easier to understand in the future.

Reply:
I agree call as a function within `ReportAfterEach` should be cleaner in case there are other similar things need to be managed in the `ReportAfterEach`. But the change in the `SynchronizedBeforeSuite`, `SynchronizedAfterSuite` are still necessary and it is trivial, we can keep it as is. WDYT?

Reply:
The `SynchronizedBeforeSuite` / `SynchronizedAfterSuite` changes should definitely be function calls into here - they set arbitrary strings on state that is then used by external tools, keeping them in one place is super useful.

Reply:
Done.